### PR TITLE
[BEAM-3304] documentation for PaneInfo in BPG

### PR DESCRIPTION
**Pull request by riteshghorse**
Update Beam Programming guide to reflect usage of PaneInfo for Go SDK.
image

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Review by damccorm — COMMENTED** [review]
Request as a reviewer - in the future for website changes, could you please link to the rendered website page that you're modifying, or take a screenshot if its a small change? Website_Stage_GCS automatically publishes a version, you can find the link in the CI build (in this case its http://apache-beam-website-pull-requests.storage.googleapis.com/17047/documentation/programming-guide/index.html#other-dofn-parameters). Personally, I like embedding a screenshot in the description even more so there's no link following needed

File: website/www/site/content/en/documentation/programming-guide.md [hunk 1]
```diff
@@ -2150,9 +2149,11 @@ class ProcessRecord(beam.DoFn):
...
 
 {{< highlight go >}}
-// PaneInfo and triggers are not yet implemented in the Go SDK.
-// See https://issues.apache.org/jira/browse/BEAM-3304 for info
-// on contributing triggers and panes.
+func MyDoFn(pn beam.PaneInfo, word string) string {
+  t := int(pn.timing)
+  isFirstPane, isLastPane := pn.IsFirst, pn.IsLast
+  ...
+}
```
[the remark this answers is not in the document]
Comment (riteshghorse) [reply website/www/site/content/en/documentation/programming-guide.md]: Done

**Review by damccorm — APPROVED** [review]
Nice! LGTM

image

**Comment by damccorm** [comment]
R: @youngoli - looks like the bot still can't authenticate 😢

**Comment by youngoli** [comment]
This looks mostly good to me, except I think the changes to Beam itself should go in a separate PR and the website changes should be submitted once the feature is available in a current release (so we should wait a little if this is coming in 2.38).

Ritesh, can you create a new PR with just the changes in pkg/beam and test/integration? That one should be a quick review & merge from me.

**Comment by riteshghorse** [comment]
The pane observation is available in 2.37.0 but I didn't the documentation at that time.

**Comment by riteshghorse** [comment]
R: @lostluck

**Review by lostluck — APPROVED** [review]
LGTM! Thanks Ritesh :D 

https://apache-beam-website-pull-requests.storage.googleapis.com/17047/documentation/programming-guide/index.html#other-dofn-parameters

**Pull request merged**